spec/unit/plugins/linux/network_spec.rb

```diff
@@ -1335,6 +1335,59 @@
         end
       end
 
+      describe "when there's a source field in a local route entry but no dev" do
```

Comment:
So if I understand, in this one:
* 172.16.19.39 is in fact on eth0, so that one is considered
* 10.116.201.254 is not on eth0, so that one is not considered
* Metric 10 is lower than metric 20, so the multipath route wins over the only other default route.

Reply:
added a clear route by route explanation of why choices are made for each of the new test cases - please take a look